sips/sip-023/sip-023-emergency-fix-traits.md

```diff
@@ -0,0 +1,152 @@
+# Preamble
+
+SIP Number: 023
+
+Title: Emergency Fix to Trait Invocation Behavior
+
+Authors:
+    Aaron Blankstein <aaron@hiro.so>,
+    Brice Dobry <brice@hiro.so>,
+    Jude Nelson <jude@stacks.org>,
+
+Consideration: Technical, Governance
+
+Type: Consensus
+
+Status: Draft
+
+Created: 1 May 2023
+
+License: BSD 2-Clause
+
+Sign-off: 
+
+Discussions-To: https://github.com/stacksgov/sips
+
+# Abstract
+
+On 1 May 2023, it was discovered that smart contracts deployed prior to Stacks 2.1
+that exposed public methods with
+trait arguments could not be invoked with previously working trait-implementing
+contract arguments.
+
+This bug was caused by the activation of Stacks Epoch 2.2 (https://github.com/stacksgov/sips/blob/main/sips/sip-022/sip-022-emergency-pox-fix.md).
+
+This SIP proposes an **immediate consensus-breaking change** to
+introduce a new Stacks epoch 2.3 that corrects this regression.
+
+**This SIP proposes a Bitcoin activation height of 788,287**
```

Comment:
This would roughly fall on Thursday 3-4pm ET. To ensure smooth activation, IMO a height where activation happens earlier in the day ET is better (in terms of availability of folks in case issues arise). Based on comments from @philipdesmedt , @ChienteH, @muneeb-ali etc, it also seems the desire is to have this window as short as safely doable.

I'd like to propose block 788,240 - which is estimated to happen approx. 8am ET on Thursday.

Reply:
upated sip to 788,240